### ISSUE TITLE
Import Brick Owl XML backup

### ISSUE THREAD

**Issue by jbcpollak** [rgriebl/brickstore#304]
Can you import a BrickStore XML backup file?

When I try, I get the error:

> Could not parse the XML data.
>
> Parse error: expected root node INVENTORY, but got inventory

**Comment by paramecie** [comment]
> a BrickStore XML

... a Brick Owl?

Yes, I know, all those Brick Something are confusing :)
PS: maybe you could upload any file so those who aren't using Owl could see?

**Comment by jbcpollak** [comment]
🤦 sorry, yes, typo fixed, and file uploaded. I also hadn't looked before but now I see that one of the tags in the file is explicitly `brickowl_item_id`, so maybe not something that was planned, or maybe its an entirely custom schema.

BTW this all came up because I was trying to do a manual bi-directional sync between BrickStore and BrickOwl. You can't set a lot id in BrickStore as far as I can tell, so I imported the brickstore file into BrickOwl, then downloaded the above XML file and was hoping to import it into BrickStore, to keep everything in sync

**Comment by paramecie** [comment]
Yes, terminology is different, example "lot" in Owl which is an "item" in BS, or "LotID" in BS and "lot_id" in Owl...

I didn't check everything but it seems not really a problem to import with some conversion.
We'll see what Robert can do?

Until then, you could use Notepad++ and set a macro replacement for those tokens, but it's some work to be done ;-)

**Comment by rgriebl** [comment]
> I didn't check everything but it seems not really a problem to import with some conversion.
> We'll see what Robert can do?

In theory - yes. In practice I'd like to see a XML schema for this format first to eliminate any guess work.

**Comment by ZZJHONS** [comment]
> > I didn't check everything but it seems not really a problem to import with some conversion.
> > We'll see what Robert can do?
> 
> In theory - yes. In practice I'd like to see a XML schema for this format first to eliminate any guess work.

Not a XML schema, i couldn't find any, but here is the API docs, on the Store Inventory tab there is a description of each field: https://www.brickowl.com/api_docs

**Comment by rgriebl** [comment]
I had the "pleasure" of looking at the bricksync source to fix #51 and I even fixed the bug right in bricksync in the end. 
But this code base is a not-invented-here-syndrome nightmare. There's no way to extract any kind of usable code.
I'm not using and not planning on using BrickOwl personally right now (neither as buyer or seller), so don't expect to see an implementation coming from my side any time soon.

**Comment by cwmcd** [comment]
fixed a bug right in bricksync? should i update bricksync or has the fix just been 'submitted' as an issue on the github?

**Comment by rgriebl** [comment]
See #345

**Comment by cwmcd** [comment]
@rgriebl nice, ty
